core/src/main/java/tc/oc/pgm/compass/targets/PlayerCompassTarget.java

```diff
@@ -0,0 +1,68 @@
+package tc.oc.pgm.compass.targets;
+
+import static tc.oc.pgm.util.player.PlayerComponent.player;
+
+import java.util.Optional;
+import net.kyori.adventure.text.Component;
+import org.bukkit.Location;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.compass.CompassTarget;
+import tc.oc.pgm.compass.CompassTargetResult;
+import tc.oc.pgm.filters.query.PlayerQuery;
+import tc.oc.pgm.util.named.NameStyle;
+
+public class PlayerCompassTarget implements CompassTarget {
+
+  private final Filter filter;
+  private final Component name;
+  private final boolean showPlayerName;
+
+  public PlayerCompassTarget(Filter filter, Component name, boolean showPlayerName) {
+    this.filter = filter;
+    this.name = name == null ? Component.translatable("compass.tracking.player") : name;
+    this.showPlayerName = showPlayerName;
+  }
+
+  @Override
+  public Filter getFilter() {
+    return filter;
+  }
+
+  @Override
+  public Component getName(Match match, MatchPlayer player) {
+    return showPlayerName ? player(player, NameStyle.FANCY) : name;
```

Comment:
pretty sure this shouldn't be the behavior, it's not an either-or kinda deal, should be possible to show both

name="FlagCarrier" & show name: `Flag carrier: Pablete1234 (34m)`
name="FlagCarrier": `Flag carrier (34m)`
just show name: `Pablete1234 (34m)`
in all of those the (34m) at the end is optional

Reply:
that's a good point